### PR TITLE
Allow User Data Consistency Test to Output Meaningful Errors--Closes Issue #17

### PR DESCRIPTION
**Pull request by SonikaG**
Hi guys! Here's my pull request--sorry it took so long (I'm a little fuzzy with git). 
There  are some things I have done in here that may need adjustment:
1. I didn't want it to print the detailed test results every time you run the program. I simply wanted to put them in the log file. Right now I am passing a boolean flag to let TestSuiteResult know if the outstream is the log file. If there's a better way of doing this please let me know. Also, if we'd like to add an option for the user to decide whether they want a log file and what they want printed to it, let me know. 
2. If there is no user description to be added to the test failure description, then that section is left blank and would look like: 
test #1 FAILED file_missing:
as opposed to 
test #1 FAILED file_missing: mnt/snapshot/test_dir/test_file is missing
Also I only supply the description if the test failed, otherwise there is no output. 

I am new to c++, so please let me know if there is a better or more efficient way of doing something. 

Thank you and please let me know if you have any questions :) 

Sonika

### CODE REVIEW

File: code/harness/c_harness.cpp [hunk 1]
```diff
@@ -669,14 +671,30 @@ int main(int argc, char** argv) {
...
+    test_harness.PrintTestStats(cout, false);
     cout << endl;
 
     for (unsigned int i = 0; i < Tester::NUM_TIME; ++i) {
       cout << "\t" << (Tester::time_stats) i << ": "
         << test_harness.get_timing_stat((Tester::time_stats) i).count() << " ms"
         << endl;
     }
+    //get the name of the test being run 
+    string path = argv[test_case_idx];
```
Comment (ashmrtn) [inline code/harness/c_harness.cpp]: Could you actually move this to the top of the file. I don't really like the idea of accessing a command line arg so far down in the file :)
Could you also make it `const string` if you don't plan on changing it? Just so it doesn't accidentally get changed later on.

File: code/harness/c_harness.cpp [hunk 2]
```diff
@@ -669,14 +671,30 @@ int main(int argc, char** argv) {
...
+    int begin = path.rfind('/');
+    //remove everything before the last /
+    string test_name = path.substr(begin +1);
```
Comment (ashmrtn) [inline code/harness/c_harness.cpp]: nit: `(begin +1)` -> `(begin + 1)`

File: code/harness/c_harness.cpp [hunk 3]
```diff
@@ -669,14 +671,30 @@ int main(int argc, char** argv) {
...
+    //remove the extension 
+    test_name = test_name.substr(0, test_name.length()-3);
```
Comment (ashmrtn) [inline code/harness/c_harness.cpp]: nit: Space around minus operator please `length() - 3)`

File: code/harness/c_harness.cpp [hunk 4]
```diff
@@ -669,14 +671,30 @@ int main(int argc, char** argv) {
...
+    //get the name of the test being run 
+    string path = argv[test_case_idx];
+    int begin = path.rfind('/');
+    //remove everything before the last /
+    string test_name = path.substr(begin +1);
+    //remove the extension 
+    test_name = test_name.substr(0, test_name.length()-3); 
+    //get the date and time stamp and format
+    time_t now = time(0); 
+    char* time = ctime(&now);
```
Comment (ashmrtn) [inline code/harness/c_harness.cpp]: `char* time` appears to be unused...?

File: code/results/TestSuiteResult.cpp [hunk 5]
```diff
@@ -40,23 +40,31 @@ void TestSuiteResult::PrintResults(ostream& os) const {
...
         case DataTestResult::kFileMetadataCorrupted:
+          des += "file metadata corrupted: ";
           ++file_metadata_corrupted;
           break;
         case DataTestResult::kOther:
+          des += "other: ";
           ++other;
           break;
       }
+      if(is_log)
```
Comment (ashmrtn) [inline code/results/TestSuiteResult.cpp]: nit: `if(is_log)` -> `if (is_log) {`

Always use curly braces and add a space character between the if, for, while, etc and parentheses please.

File: code/results/TestSuiteResult.cpp [hunk 6]
```diff
@@ -40,23 +40,31 @@ void TestSuiteResult::PrintResults(ostream& os) const {
...
+          os << "Test #" << num_failed << " FAILED: "<< des << result.data_test.error_description <<"\n";
```
Comment (ashmrtn) [inline code/results/TestSuiteResult.cpp]: nit: spaces between `<<` operators and fields please

`num_failed` isn't necessarily the absolute index of the test that failed, just the count of the tests that have failed so far. You will need another counter or you will need to modify the for loop to be index-based.

File: code/harness/c_harness.cpp [hunk 7]
```diff
@@ -7,7 +7,9 @@
 #include <unistd.h>
 #include <wait.h>
 
+#include <ctime>
 #include <iostream>
+#include <fstream>
```
Comment (vijay03) [inline code/harness/c_harness.cpp]: Nit: lets keep the include ordered alphabetically.

**Comment by vijay03** [comment]
@SonikaG could you also add a comment showing what the output actually looks like for some test run?

**Comment by SonikaG** [comment]
I've fixed the nits and styling issues. Here's an example log file. Let me know if anything needs to be changed
[20170920_01_12_33-echo_sub_dir.txt](https://github.com/utsaslab/crashmonkey/files/1320018/20170920_01_12_33-echo_sub_dir.txt)

**Comment by vijay03** [comment]
@SonikaG, please go ahead and merge the pull request into master. Could you rename the pull request so that it is a bit more descriptive?

**Pull request merged**